src/OrleansAzureUtils/Providers/Storage/AzureTableStorage.cs

```diff
@@ -234,9 +247,11 @@ internal void ConvertToStorageFormat(object grainState, GrainStateEntity entity)
 
                 if (Log.IsVerbose3) Log.Verbose3("Writing JSON data size = {0} for grain id = Partition={1} / Row={2}",
                     data.Length, entity.PartitionKey, entity.RowKey);
-                
-                dataSize = data.Length;
-                entity.StringData = data;
+
+                dataSize = data.Length * 2;
```

Comment:
why is that multiplied by 2? Can you also add a comment in the code explaining that.